pwnlib/libcdb.py

```diff
@@ -126,7 +131,25 @@ def provider_local_system(hex_encoded_id, hash_type):
         return local_libc.data
     return None
 
-PROVIDERS = [provider_local_system, provider_libcdb, provider_libc_rip]
+# Offline search https://github.com/niklasb/libc-database for hash type
+def provider_local_database(hex_encoded_id, hash_type):
+    if not context.local_libcdb:
+        return None
+
+    localdb = Path(context.local_libcdb)
+    if not localdb.is_dir():
+        if context.local_libcdb != DEFAULT_LOCAL_LIBCDB:
+            log.warn_once("%s does not exist, please download libc-database first.", context.local_libcdb)
```

Comment:
As for the warning, I would probably like it better in the validator for local libc (when the user sets it) rather than when it is needed. It would then automatically be sent only once per location, and visible even of the location points to the default path.

Reply:
I don't think the validator is run when parsing the `pwn.conf` or the LOCAL_LIBCDB argument/environment var. Or are they run when setting the defaults too?

https://github.com/Gallopsled/pwntools/blob/000c31c15527601a7f4ff5f39d2688d362fe3a40/pwnlib/context/__init__.py#L1630

Reply:
I've modified the HOOK function, now the validator can run while parsing the LOCAL_LIBCDB environment variable. Actually, it is also work when parsing the `pwn.conf`.

Reply:
Ah, you are right. Now the doctests need fixing and we can 🚢